plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -45,18 +50,74 @@ func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(usernam
 		glog.Info("Failed: Starting openstack authenticate client")
 		return nil, false, errors.New("Failed to authenticate")
 	}
-
+//FIXME user_id, not user_name...
 	return &user.DefaultInfo{Name: username}, true, nil
 }
 
-// NewKeystoneAuthenticator returns a password authenticator that validates credentials using openstack keystone
-func NewKeystoneAuthenticator(authURL string) (*KeystoneAuthenticator, error) {
-	if !strings.HasPrefix(authURL, "https") {
-		return nil, errors.New("Auth URL should be secure and start with https")
+func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticateToken(value string) (user.Info, bool, error) {
```

Comment:
If we are to keep it, it starts getting complicated from here I think.

We can only ever support one domain with username/password on the server. Domain can be specified in the config though, so we could read it from the config object and put it into the gophercloud.AuthOptions, but probably only should conditionally if the keystone endpoint is v3? Its doable, but a lot of code for a feature we're not sure is needed?

The username/password feature is still marked experimental so removing it now would be really ideal if we believe tokens is a better approach since no one should be relying on it yet.

I'm not sure how authz will play well with username/password either. I'm still working through those details on the token side. I think for Magnum's use case we want to ensure that access is restricted to a particular tenant, so one tenant can't login to another tenants k8s.